### PR TITLE
Support for Adding Additional Reserved IP to Existing Linodes 

### PR DESCRIPTION
**Pull request by AniJ98**
## 📝 Description

This PR introduces functionality and tests for creating Linodes with reserved IP addresses and adding additional reserved IPs to existing Linodes in the Linodego client. These changes are necessary to provide comprehensive support for managing Linodes with reserved IP addresses through the Linode API. The changes include:

1. Implementation of core operations:

- Creating a Linode with a reserved IP
- Adding an additional reserved IP to an existing Linode

2. Test coverage:

- TestInstance_CreateWithReservedIPAddress: Tests creating a Linode with a reserved IP.
- TestInstance_CreateWithReservedIPAddress (Variants): Tests various scenarios for creating a Linode with a reserved IP.
- TestInstance_AddReservedIPToInstance: Tests adding a reserved IP to an existing Linode.
- TestInstance_AddReservedIPToInstanceVariants: Covers different cases for adding a reserved IP to an existing Linode.

## ✔️ How to Test

What are the steps to reproduce the issue or verify the changes?

1. Ensure you have a valid Linode API token. Set up your environment: 

 `export LINODE_TOKEN="your_token_here" `

Additionally you need to set the following if you want to test it in a different environment:

  ```
  export LINODE_URL="https://api.linode.com"
  export LINODE_API_VERSION="v4beta"
  ```

2. Navigate to the test directory within the linodego project. Update the LINODE_TOKEN in the Makefile: 

  `LINODE_TOKEN="your_token_here" `

  Run the tests using one of the following commands: To run all integration tests:
  
  `make testint`
  
  To run a specific test: 
  
  `go test -v ./integration -run TestInstance_CreateWithReservedIPAddress`
  
  Verify the test output for any failures or unexpected behavior.

Note:
Ensure that you have enough quota to reserve IP before adding it to an existing linode. Ensure the IPMAX limit is enough to add additional reserved IP addresses to the linode.

### CODE REVIEW

File: instance_ips.go [hunk 1]
```diff
@@ -144,3 +150,13 @@ func (c *Client) DeleteInstanceIPAddress(ctx context.Context, linodeID int, ipAd
 	err := doDELETERequest(ctx, c, e)
 	return err
 }
+
+// Function to add additional reserved IPV4 addresses to an existing linode
+func (c *Client) AddReservedIPToInstance(ctx context.Context, linodeID int, opts InstanceReserveIPOptions) (*InstanceIP, error) {
```
Comment (lgarber-akamai) [inline instance_ips.go:155]: nitpick: Could this endpoint method be renamed to `AssignInstanceReservedIP` for consistency?

File: instances.go [hunk 2]
```diff
@@ -164,6 +164,8 @@ type InstanceCreateOptions struct {
 
 	// Deprecated: group is a deprecated property denoting a group label for the Linode.
 	Group string `json:"group,omitempty"`
+
+	Ipv4 []string `json:"ipv4,omitempty"`
```
Comment (lgarber-akamai) [inline instances.go:168]: nitpick:
```suggestion
	IPv4 []string `json:"ipv4,omitempty"`
```

File: instance_ips.go (hunk 1 above)
Comment (AniJ98) [reply instance_ips.go]: Changed the name to AssignInstanceReservedIP

File: instances.go (hunk 2 above)
Comment (AniJ98) [reply instances.go]: Made this change too

**Review by lgarber-akamai — APPROVED** [review]
All tests are passing on my end. Thanks for the quality contribution and the comprehensive test suite!

**Review by zliang-akamai — APPROVED** [review]
Looks good to me!

**Pull request merged**